doc: add explanation for multiple `Transfer-Encoding` values (#39496)
mdn/content: files/en-us/web/http/reference/headers/transfer-encoding/index.md

Each segment of a multi-node connection can use different `Transfer-Encoding` values.
If you want to compress data over the whole connection, use the end-to-end {{HTTPHeader("Content-Encoding")}} header instead.

> When present in a request it indicates the encodings supported by the client.

Reviewer: Not quite, the `TE` header is supposed to be used for this, see https://datatracker.ietf.org/doc/html/rfc9112#section-7.4

Reviewer 2: Thanks for finding that! I have tried to update this bit in https://github.com/mdn/content/pull/39496/files#r2114994600

> When present in a request it indicates the encodings applied to the request body.

Reviewer: ```suggestion
In requests, it indicates the encodings applied to the request body.
```

> When present in a request it indicates the encodings supported by the client.
> Multiple encodings may be specified, and the client must support the `chunked` directive.

Reviewer 2: ```suggestion
When present in a request it indicates the encodings applied to the request body.
While multiple encodings may be specified, and the client must support the `chunked` directive, in practice this header is rarely used in requests (and if used, only with `chunked`).
```

> When present on a response it indicates the compression used on the message, and/or whether the message has been chunked.

Reviewer: ```suggestion
In responses, it indicates the compression used on the message, and/or whether the message has been chunked.
```

Note that if the message is chunked, this must be applied last, after any other compression.
For example, `Transfer-Encoding: gzip, chunked` indicates that the content has been compressed using the gzip coding and then chunked using the chunked coding while forming the message body.
While multiple compression options may be specified in theory, there is no practical value in applying more than one compression.

The response to a {{HTTPMethod("HEAD")}} request has no body and no transfer encoding, so the header is optional.

> When present it indicates the value that would have applied to the corresponding response to a {{HTTPMethod("GET")}} message, if that `GET` request did not include a preferred `Transfer-Encoding`.

Reviewer 2: I agree in principle, but my initial problem with this is that it talked about responses only, so it looked like we had omitted one case. I think this is good but I'll merge and see the shape of it.
